src/router/router.rs

```diff
@@ -159,23 +159,31 @@ fn creates_valid_regex_for_routes () {
     assert_eq!(regex1.is_match("foo/4711/bar?foo=true&bar=false"), false);
     assert_eq!(regex1.is_match("foo/4711/bar/test%20spacing"), true);
     assert_eq!(regex1.is_match("foo/4711/bar/5281?foo=test%20spacing&bar=false"), true);
+    assert_eq!(regex1.is_match("foo/alice/bar/bob"), true);
 
     assert_eq!(regex2.is_match("foo/4711/bar"), true);
     assert_eq!(regex2.is_match("foo/4711/barr"), false);
     assert_eq!(regex2.is_match("foo/4711/bar?foo=true&bar=false"), true);
     assert_eq!(regex2.is_match("foo/4711/4712/bar"), false);
     assert_eq!(regex2.is_match("foo/4711/4712/bar?foo=true&bar=false"), false);
+    assert_eq!(regex2.is_match("foo/alice/bar/bob"), false);
+    //assert_eq!(regex1.is_match("foo/alice/bar"), true);
```

Comment:
Good spot. What a noob mistake ^-^

Good idea to assert the data captured too. Want me to do this too?

On 14 April 2015 at 13:07, Ryman <notifications@github.com> wrote:

> In src/router/router.rs
> <https://github.com/nickel-org/nickel.rs/pull/190#discussion_r28320639>:
>
> >
> >      assert_eq!(regex2.is_match("foo/4711/bar"), true);
> >      assert_eq!(regex2.is_match("foo/4711/barr"), false);
> >      assert_eq!(regex2.is_match("foo/4711/bar?foo=true&bar=false"), true);
> >      assert_eq!(regex2.is_match("foo/4711/4712/bar"), false);
> >      assert_eq!(regex2.is_match("foo/4711/4712/bar?foo=true&bar=false"), false);
> > +    assert_eq!(regex2.is_match("foo/alice/bar/bob"), false);
> > +    //assert_eq!(regex1.is_match("foo/alice/bar"), true);
>
> Should it be regex2.is_match(..) rather than regex1?
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/nickel-org/nickel.rs/pull/190/files#r28320639>.
>